.github/workflows/test.yml

```diff
@@ -1,15 +1,20 @@
 name: Test
 
-on: [push, pull_request]
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
 
 jobs:
   test:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # 2.3.5
-      - uses: actions/setup-dotnet@9211491ffb35dd6a6657ca4f45d43dfe6e97c829 # 1.7.2
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+      - uses: actions/setup-dotnet@9211491ffb35dd6a6657ca4f45d43dfe6e97c829
         with:
           dotnet-version: "6.0.100"
       - run: pwsh ./test.ps1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
For understanding, is this here because checkout used to need it and now no longer does. Or was it never needed for this task?

Reply:
It was never need. I think it was just a copy/paste error.